README.md
# vuecli-muse-admin-template

English | [简体中文](./README_CN.md)

---

## Introduction

vuecli-muse-admin-template is a background integration solution. It is created by [vue-cli 3](https://cli.vuejs.org/), uses the UI toolkit [Muse-UI](https://muse-ui.org/), and aimed at helping to build a complex Single-Page Application.

See [Features](#Features) to get more.

**Improving it...** :D

## Preparation

- Install [Node.js](https://nodejs.org/en/) and [git](https://git-scm.com/) locally.

- This project is based on ES2015+, [Vue](https://vuejs.org/), [Vuex](https://vuex.vuejs.org/), [VueRouter](https://router.vuejs.org/guide/), [Axios](https://github.com/axios/axios) and [Muse-UI](https://muse-ui.org/).

- All request data is simulated using [vuecli-muse-admin-template-server](https://github.com/ModyQyW/vuecli-muse-admin-template-server).

- Understanding what listed above in advance will greatly help use this template.

## Getting Started

```sh
# install
# npm i
yarn

# compiles and hot-reloads for development
# npm run serve
yarn serve

# run tests
# npm run test
yarn test

# Compiles and minifies for production
# npm run build
yarn build

# Lints and fixes files
# npm run lint
yarn lint
```

**Attention**: I import local material-design-icons related files in [@/plugins/muse/index.js](./src/plugins/muse/index.js) for now cause using npm or yarn to install always leads to a error in China.

if you want to use npm or yarn to install material-design-icons, follow the steps below.

- comment Line 63 in [@/plugins/muse/index.js](./src/plugins/muse/index.js)
- run `npm i --save material-design-icons` or `yarn add material-design-icons` in shell
- uncomment Line 64 in [@/plugins/muse/index.js](./src/plugins/muse/index.js)

**Do not** use unpkg's and bootcss's css of material icons. They will lead to lack of some icons.

## Features

- [Customize Configuration CLI](https://cli.vuejs.org/config/)

- Access Control

|Level|Role|
|:-:|:--|
|0|super admin|
|1|admin|
|2|manager|
|3|developer|
|4|client|
|5|user|

- global

  - Two environments for building: dev & prod

  - Two languages: English and Chinese

  - Dynamic sidebar

  - Dynamic breadcrumb

- Clip to copy

- Counter

- Zip export

- Draggable

- Icons

- Editor

  - Rich text editor

  - Markdown editor

  - JSON editor

- Simple table

- Error pages: 401, 403, 404, 500

- Charts: vue-echarts

## Code Style

Please follow [Vue.js Style Guide](https://vuejs.org/v2/style-guide/index.html).

## Usage

See [usage.md](./docs/usage.md).

## Changelog

See [changelog.md](./docs/changelog.md).

## Online Demo

[demo](https://modyqyw.top/vuecli-muse-admin-template/dist/#/signin)

## Donate

I am still working on it to imporve - not only add new function but also complete codes and documents.

If you think this is helpful, you can buy me a coffee :D

|WeChat|Alipay|
|:-:|:-:|
|<img src="https://raw.githubusercontent.com/ModyQyW/Pictures/master/wechat.png" style="width: 300px;"/>|<img src="https://raw.githubusercontent.com/ModyQyW/Pictures/master/alipay.jpg" style="width: 300px;" />|

## Browsers Support

Expect to run in latest Chrome and lastest Firefox.

Good luck to IE, Edge and Safari.

Testing...

## License

MIT

docs/changelog.md
# Changelog

## 2019/03/18

- Add：Add documents. Most documents are complete.

## 2019/03/15

- Open source.

src/router/routes.js
import dashboard from './modules/dashboard';
import nested from './modules/nested';
import clipboard from './modules/clipboard';
import zip from './modules/zip';
import editor from './modules/editor';
import draggable from './modules/draggable';
import table from './modules/table';
import icons from './modules/icons';
import customScatterChart from './modules/customScatterChart';

/*
 * Attention: the route must redirect to its child if it has children.
 * Please check ./modules/dashboard.js and ./modules/nested.js
 * 注意: 假如路由有 children 字段, 必须重定向到其 child
 * 请参考 ./modules/dashboard.js 和 ./modules/nested.js
 *
 * meta: {
 *   cached: <Boolean>, keep alive or not, default false,
 *   defaultOpen: <Boolean>, default open or not if nested, default false,
 *   hidden: <Boolean>, not show in drawer or not, default false,
 *   icon: <String>, meterial design icon name, default showing nothing,
 *                   no need to set if show in drawer or need no icon
 *   role: <Number>, minimal role for accessing the route, default all roles,
 *   title: <String>, the name show in drawer, will be localized,
 *                    if not set, use the title of the first child
 * }
 *
 * meta: {
 *   cached: <Boolean>, 是否保持组件状态, 默认为否,
 *   defaultOpen: <Boolean>, 存在嵌套项时是否默认展开, 默认为折叠,
 *   hidden: <Boolean>, 是否不在抽屉导航中显示, 默认为否，
 *   icon: <String>, meterial design 图标名称, 默认不显示图标,
 *                   如果不在抽屉导航中显示或不需要图标, 不用设置,
 *   role: <Number>, 路由允许进入的最低角色, 默认所有角色均能进入,
 *   title: <String>, 显示在抽屉导航中的名称, 会进行国际化,
 *                    如果没有设置, 会使用第一个 child 的 title
 * }
 */

// routes dont need token
// 不需要 token 的 routes
export const commonRoutes = [
  '/',
  '/signin',
];

export const exceptionRoutes = [
  '/401', '/403', '/404', '/500',
];

// routes every role has
// 每个角色都能访问的 routes
export const constRoutes = [
  {
    path: '/',
    redirect: '/signin',
    meta: {
      hidden: true,
    },
  },
  {
    name: 'signin',
    path: '/signin',
    component: () => import('@/views/signin/index'),
    // component: () => import('@/views/layout/index'),
    meta: {
      hidden: true,
    },
  },
  {
    name: '401',
    path: '/401',
    component: () => import('@/views/exception/401'),
    meta: {
      hidden: true,
    },
  },
  {
    name: '403',
    path: '/403',
    component: () => import('@/views/exception/403'),
    meta: {
      hidden: true,
    },
  },
  {
    name: '404',
    path: '/404',
    component: () => import('@/views/exception/404'),
    meta: {
      hidden: true,
    },
  },
  {
    name: '500',
    path: '/500',
    component: () => import('@/views/exception/500'),
    meta: {
      hidden: true,
    },
  },
];

// routes for special role
// 特定角色才能访问的 routes
export const mutableRoutes = [
  // dashboard
  // 仪表板
  dashboard,
  // nested routes
  // 嵌套路由
  nested,
  // clipboard
  // 剪切板
  clipboard,
  // zip
  zip,
  // editor
  // 编辑器
  editor,
  // draggable
  // 拖拽
  draggable,
  // table
  // 表格
  table,
  // icons
  // 图标
  icons,
  // custom scatter chart
  // 自定义散点图
  customScatterChart,
  // 404
  {
    path: '*',
    redirect: { name: '404' },
    meta: {
      hidden: true,
    },
  },
];
